ts/request.ts

```diff
@@ -18,6 +18,10 @@ request.send(function(err, data) {
 request.on('error', function(err, response) {
 
 });
+request.onAsync('validate', function(request, done) {
+  console.log(request.httpRequest.body);
+  if (done) done()
```

Comment:
This is the pattern that most likely will be necessary, with `doneCallback?` annotated as optional. It might be better to make the param required so that the `if` check will not be necessary (depending on the typescript config).